sdk/how-to-guides/ports.md

```diff
@@ -0,0 +1,144 @@
+> See first: [Juju `open-port`](https://juju.is/docs/juju/hook-tool#open-port)
+
+**Contents**:
+
+1. [Open a port based on the charm config](#heading--open-a-port-to-your-charm)
+1. [Test the port is opened](#heading--test-the-port-is-opened)
+
+<a href="#heading--heading--open-a-port-to-your-charm"><h2 id="heading--heading--open-a-port-to-your-charm">Open a port based on the charm config</h2></a>
+
+To make a service permanently reachable under a stable URL on the cluster the charm needs to open (expose) a port. This ensures that the charm will be consistently accessible even if the pod gets recycled and the IP address changes.
+
+The port that the charm's service should be reachable on is typically a config option; for example, by including a section like this in `charmcraft.yaml`:
+
```

Comment:
Rephrase suggestion:

(a) to improve clarity:
 - 
> typically defined as a config option

(b) to align with the order in which people need the info:

> for example, by having your `charmcraft.yaml` include a section like this: